library_generation/test/integration_tests.py

```diff
@@ -0,0 +1,148 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import shutil
+import unittest
+from distutils.dir_util import copy_tree
+from distutils.file_util import copy_file
+from filecmp import dircmp
+
+from lxml import etree
+from git import Repo
+from pathlib import Path
+from typing import List
+from typing import Dict
+from library_generation.generate_repo import generate_from_yaml
+from library_generation.model.generation_config import from_yaml
+from library_generation.test.compare_poms import compare_xml
+from library_generation.utilities import get_library_name
+
+config_name = "generation_config.yaml"
+script_dir = os.path.dirname(os.path.realpath(__file__))
+# for simplicity, the configuration files should be in a relative directory
+# within config_dir named {repo}/generation_config.yaml, where repo is
+# the name of the repository the target libraries live.
+config_dir = f"{script_dir}/resources/integration"
+golden_dir = f"{config_dir}/golden"
+repo_prefix = "https://github.com/googleapis"
+committish_list = ["chore/test-hermetic-build"]  # google-cloud-java
+
+
+class IntegrationTest(unittest.TestCase):
+    def test_generate_repo(self):
+        shutil.rmtree(f"{golden_dir}", ignore_errors=True)
+        os.makedirs(f"{golden_dir}", exist_ok=True)
+        config_files = self.__get_config_files(config_dir)
+        i = 0
+        for repo, config_file in config_files.items():
+            repo_dest = f"{golden_dir}/{repo}"
+            self.__pull_repo_to(Path(repo_dest), repo, committish_list[i])
+            library_names = self.__get_library_names_from_config(config_file)
+            # prepare golden files
+            for library_name in library_names:
+                copy_tree(f"{repo_dest}/{library_name}", f"{golden_dir}/{library_name}")
+            copy_tree(
+                f"{repo_dest}/gapic-libraries-bom", f"{golden_dir}/gapic-libraries-bom"
+            )
+            copy_file(f"{repo_dest}/pom.xml", golden_dir)
+            generate_from_yaml(
+                generation_config_yaml=config_file, repository_path=repo_dest
+            )
+            # compare result
+            for library_name in library_names:
+                print(
+                    f"Compare generation result: "
+                    f"expected library in {golden_dir}/{library_name}, "
+                    f"actual library in {repo_dest}/{library_name}."
+                )
+                compare_result = dircmp(
+                    f"{golden_dir}/{library_name}",
+                    f"{repo_dest}/{library_name}",
+                    ignore=[".repo-metadata.json"],
+                )
+                # compare source code
+                self.assertEqual([], compare_result.left_only)
+                self.assertEqual([], compare_result.right_only)
+                # self.assertEqual([], compare_result.diff_files)
```

Comment:
The transport section in README.md is not rendered correctly because the value in template is grpc, http and both, rather than grpc, rest and grpc+rest.

PR to fix this issue: https://github.com/googleapis/synthtool/pull/1925